src/plugins/createPluginManager.ts

```diff
@@ -0,0 +1,35 @@
+import {Plugin, PluginItems} from '@plugins/PluginsContext';
+
+export class PluginManager {
+  private plugins: Plugin[];
+
+  public constructor() {
+    this.plugins = [];
+  }
+
+  public add(plugin: Plugin): void {
+    const order = plugin.order ?? -Infinity;
```

Comment:
Optional: Alternatively, such logic could be extracted to some `OrderedList`, so the same logic could be applied in `insertPluginsToArray`